RFC-005-SWAP-Bitcoin-basic.md

```diff
@@ -0,0 +1,248 @@
+    OP_DUP OP_HASH160 <refund_identity>
+OP_ENDIF
+OP_EQUALVERIFY
+OP_CHECKSIG
+```
+
+As required by [RFC003](./RFC-003-SWAP-basic.md), this HTLC uses absolute time locks to check whether `expiry` has been reached.
+Specifically, `OP_CHECKLOCKTIMEVERIFY` (see [BIP65](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki)) is used to compare the time in the block's header to the `expiry` in the contract.
+
+Implementations MUST consider an `expiry` value below `500000000` for a Bitcoin HTLC to be invalid due to the [`lock_time`](https://en.bitcoin.it/wiki/Protocol_documentation#tx) transaction field interpreting values below `500000000` as block heights.
+
+To compute the exact Bitcoin script bytes of the contract, implementations should use the following offset table:
+
+| Data              | Position of first byte | Position of last byte | Length (bytes) | Description                                                                |
+|:------------------|:-----------------------|:----------------------|:---------------|:---------------------------------------------------------------------------|
+| `6382012088a820`  | 0                      | 6                     | 7              | `OP_IF OP_SIZE 32 OP_EQUALVERIFY OP_SHA256` + `PUSH32` for the secret_hash |
+| `secret_hash`     | 7                      | 39                    | 32             | See [Parameters](#parameters)                                              |
+| `8876a9`          | 40                     | 42                    | 3              | `OP_EQUALVERIFY OP_DUP OP_HASH160`                                         |
+| `redeem_identity` | 43                     | 64                    | 20             | See [Parameters](#parameters)                                              |
+| `67`              | 65                     | 65                    | 1              | `OP_ELSE`                                                                  |
+| `expiry`          | 66                     | 69                    | 4              | See [Parameters](#parameters)                                              |
+| `b17576a9`        | 70                     | 73                    | 4              | `OP_CHECKLOCKTIMEVERIFY OP_DROP OP_DUP OP_HASH160`                         |
+| `refund_identity` | 74                     | 93                    | 20             | See [Parameters](#parameters)                                              |
+| `6888ac`          | 94                     | 96                    | 3              | `OP_ENDIF OP_EQUALVERIFY OP_CHECKSIG`                                      |
+
+
+## Execution Phase
+
+The following section describes how both parties should interact with the Bitcoin blockchain during the [RFC003 execution phase](./RFC-003-SWAP-basic.md#execution-phase).
+
+### Deployment
+
+At the start of the deployment stage, both parties compile the contract into bytes as described in the previous section.
```

Comment:
```suggestion
At the start of the deployment stage, both parties construct the contract into bytes as described in the previous section.
```
See https://github.com/comit-network/RFCs/pull/38#discussion_r265379886

Moreover, is it really `bytes` as it is always kept in hex representation? Only bitcoind does the translation hex -> byte.

Reply:
I think compiles is the right word because we're compiling a contract from a contract template + parameters. I removed the the thing about "into bytes", because as you say we don't want to imply a particular representation it should compile into.